lib/iris/analysis/__init__.py

```diff
@@ -592,6 +593,12 @@ def aggregate(self, data, axis, **kwargs):
         ):
             fraction_not_missing = data.count(axis=axis) / data.shape[axis]
             mask_update = 1 - mdtol > fraction_not_missing
+            if np.array(result).ndim > np.array(mask_update).ndim:
```

Comment:
```suggestion
            mask_update = np.array(1 - mdtol > fraction_not_missing)
            if np.array(result).ndim > mask_update.ndim:
```

Making `mask_update` an array will ensure it also has `reshape` and `shape` used below.

Reply:
https://github.com/SciTools/iris/pull/4755/commits/7af47a14fce0bad6269c234ece83c43b5f0d359f